docs/contributing/advanced-contributing.md

```diff
@@ -0,0 +1,141 @@
+# Advanced Contribution
+
+The contributions described below are advanced because they use complex workflows or require more experience.
+If you like the challenge, this is the place for you.
+
+By now you should have worked through the [previous parts](README.md) of the project contributing guide, and also [made at least one contribution](who-written-for.md).
+
+## About the Advanced Contribution Process
+
+This is a large project which gets many design proposals.
+Design proposal discussions can take a long time.
+
+Making a PR with a design proposal simplifies this process:
+
+* Each design proposal has its own line of comments
+* Replies around each line are easy to track
+* As proposals change, pages reset as their line items resolve
+* GitHub maintains the entire history
+
+## Refactor or Cleanup Proposal
+
+A refactor or cleanup proposal changes Docker's internal structure without altering the external behavior. To make this type of proposal:
+
+1. Fork `moby/moby`.
+
+2. Create a feature-branch and make your changes in it.
+
+3. Sync and rebase with `master`.
+
+4. Run the full test suite.
+
+5. Submit a pull request (PR):
+
+    PR's title's format:
+
+    **Refactor:** _short title_
```

Comment:
Our current convention is `area/of/code: short title` or `Short title` if it's further-reaching than one section of the codebase.